src/analysis/scoping_analysis.cpp

```diff
@@ -200,26 +210,26 @@ class ScopeInfoBase : public ScopeInfo {
 
     bool passesThroughClosure() override { return usage->passthrough_accesses.size() > 0 && !createsClosure(); }
 
-    bool refersToGlobal(InternedString name) override {
+    bool refersToGlobal(InternedString name) {
         // HAX
         if (isCompilerCreatedName(name))
             return false;
 
         if (usage->forced_globals.count(name))
             return true;
-        if (usesNameLookup)
+        if (usesNameLookup_)
             return false;
         return usage->written.count(name) == 0 && usage->got_from_closure.count(name) == 0;
     }
-    bool refersToClosure(InternedString name) override {
+    bool refersToClosure(InternedString name) {
         // HAX
         if (isCompilerCreatedName(name))
             return false;
         return usage->got_from_closure.count(name) != 0;
     }
-    bool saveInClosure(InternedString name) override {
+    bool saveInClosure(InternedString name) {
         // HAX
-        if (isCompilerCreatedName(name) || usesNameLookup)
+        if (isCompilerCreatedName(name) || usesNameLookup_)
             return false;
         return usage->referenced_from_nested.count(name) != 0;
     }
```

Comment:
Can we kill these functions?  It looks like the only use is in getScopeTypeOfName, but we could just inline these in there and it would probably be more clear exactly what checks are happening (a lot of the paths that would result from inlining these functions into getScopeTypeOfName would be dead).